src/radio_payload_cats.c

```diff
@@ -0,0 +1,51 @@
+#include <stdint.h>
+#include <stdlib.h>
+
+#include "config.h"
+#include "telemetry.h"
+#include "payload.h"
+#include "log.h"
+#include "codecs/cats/cats.h"
+#include "codecs/cats/whisker.h"
+
+#define CATS_PREAMBLE_BYTE 0x55
+#define CATS_PREAMBLE_LENGTH 4
+#define CATS_SYNC_WORD 0xABCDEF12
+#define CATS_SYNC_WORD_LENGTH 4
+
+uint16_t radio_cats_encode(uint8_t *payload, uint16_t length, telemetry_data *telemetry_data, char *message)
+{
+    uint8_t *cur = payload;
+
+    // preamble
+    for(int i = 0; i < CATS_PREAMBLE_LENGTH; i++) {
+        *(cur++) = CATS_PREAMBLE_BYTE;
+    }
+
+    // sync word
+    for(int i = CATS_SYNC_WORD_LENGTH - 1; i >= 0; i--) {
+        *(cur++) = (CATS_SYNC_WORD >> (i * 8));
+    }
+
+    uint8_t *data = malloc(length);
```

Comment:
Do we have any idea on what the maximum packet size is? It's easy to run out of memory here.